bf/bf.c

```diff
@@ -29,48 +27,78 @@ int main(int argc, char const *argv[])
 	else
 	{
 		FILE *bf_file = fopen(argv[1], "r");
-		FILE *c_output = fopen(argv[2], "w");
+        	int matches = 0;
 		char c;
-
-		fputs("/* generated by bf.c */#include <stdio.h>\nint main(void) {char s[30000]={0};char*sp=s;", c_output);
+		size_t size = 1;
+		char *op_buffer = malloc(EIGHT_KB);
+		assert(op_buffer != NULL);
 
 		while ((c = fgetc(bf_file)) != EOF)
 		{
+			if (size % EIGHT_KB)
```

Comment:
That will still fail in most practical circumstances, since the chances of `size` being a multiple of `EIGHT_KB` are pretty small. What's more likely is that `size` will go from being less than `EIGHT_KB` to more than `EIGHT_KB` in a single iteration, failing to trigger the reallocation and causing a buffer overwrite.